assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -309,6 +378,54 @@ const baseResolvers = {
 			yield fetchGetModulesStore.actions.fetchGetModules();
 		}
 	},
+
+	*canActivateModule( slug ) {
+		const registry = yield Data.commonActions.getRegistry();
+		const module = registry.select( STORE_NAME ).getModule( slug );
+
+		// We use yield dataControls.resolveSelect in favour of registry.select
+		// to avoid a race condition where the module hasn't loaded yet.
+		// @TODO after #1769, we can do this.
+		// const module = yield dataControls.resolveSelect( STORE_NAME, 'getModule', slug );
+
+		// Return `undefined` if module with this slug isn't loaded yet.
+		if ( module === undefined ) {
+			return undefined;
+		}
+
+		// A module with this slug couldn't be found return null
+		if ( module === null ) {
+			return null;
+		}
+
+		const inactiveModules = [];
+
+		module.dependencies.forEach( ( dependencySlug ) => {
+			const dependedentModule = registry.select( STORE_NAME ).getModule( dependencySlug );
+			if ( ! dependedentModule.active ) {
+				inactiveModules.push( dependedentModule.name );
+			}
+		} );
+
+		// If we have inactive dependencies, there's no need to check if we can
+		// activate the module until the dependencies have been activated.
+		if ( inactiveModules.length ) {
+			const formatter = new Intl.ListFormat( getLocale(), { style: 'long', type: 'conjunction' } );
+
+			/* translators: Error message text. 1: A flattened list of module names. 2: A module name. */
+			const errorMessage = sprintf( __( 'You need to set up %1$s to gain access to %2$s.', 'google-site-kit' ), formatter.format( inactiveModules ), module.name );
+
+			yield baseActions.receiveCheckRequirementsError( { [ slug ]: errorMessage } );
+		} else {
+			const checkResult = yield Data.commonActions.await( module.checkRequirements() );
+
+			if ( checkResult === true ) {
+				yield baseActions.receiveCheckRequirementsSuccess( slug );
+			} else {
+				yield baseActions.receiveCheckRequirementsError( checkResult );
```

Comment:
If not `true`, `checkResult` should be expected to be an error object instead of a map that contains the module slug (which would be redundant to provide anyway). In other words, in combination with my other comment above this line would need to be as follows:
```suggestion
				yield baseActions.receiveCheckRequirementsError( slug, checkResult );
```
(where `checkResult` is an `error`)

Via https://github.com/google/site-kit-wp/issues/2130#issuecomment-724225862, this should be modified to rely on errors being thrown instead of returned.

Reply:
Updated.